components/tidb_query_executors/src/fast_hash_aggr_executor.rs

```diff
@@ -416,20 +464,24 @@ where
             Some(offset) => {
                 // Group exists, use the offset of existing group.
                 states_offset_each_logical_row.push(*offset);
+                // Track sizeof(*offset)
+                n_bytes += size_of::<usize>();
             }
             None => {
                 // Group does not exist, prepare groups.
                 let offset = states.len();
                 states_offset_each_logical_row.push(offset);
                 group.insert(val, offset);
+                n_bytes += size_of::<Result<Option<S>>>() + size_of::<usize>();
                 for aggr_fn in aggr_fns {
                     states.push(aggr_fn.create_state());
+                    n_bytes += size_of::<Box<dyn AggrFunctionState>>();
```

Comment:
`size_of::<Box<dyn AggrFunctionState>>` does not include its size on the heap. `std::mem::size_of_val(&*state)` can be used to get the heap space.

Reply:
I've included the heap length too in the calculation.